### DISCUSSION TITLE
Validation alerts on page load are not read

### DISCUSSION THREAD

**Discussion by MrGamble2** [nvaccess/nvda#16556] [General]
As part of a government requirement I have been updating a website to meet standards. We run ASP.NET and when you submit a page and backend validation fails (can't find account during a login, account locked, unexpected server outage, etc) the page is actually reloading and errors are just passed with the model and displayed as statically loaded elements. I believe this to be relevant discussion to the issue I am facing: https://github.com/w3c/aria/issues/1360 

I would think that like JAWS, NVDA should support reading role="alert" on form load so developers don't have to add extra JS to get the error to read for NVDA. I'm changing our implementation for now but was curious if I am just missing something here or if its a feature worth discussing. Thanks!

**Comment by XLTechie** [comment]
@MrGamble2, Is it possible for you to provide an example page which demonstrates this behavior?
Use codepen or jsfiddle or the like.
Though I believe this has been discussed before.

cc @jcsteh @aleventhal

**Reply by MrGamble2** [reply]
Hey, so something as simple as https://jsfiddle.net/veon2x1m/ would be read by jaws. (though it isnt when in jsfiddle for some reason)  Maybe its just a quirk of JAWS and it doesn't see the asp.net reload as a new page? Someone tested in the linked aria thread so it should work on a basic page load like jsfiddle as well I would think.

**Comment by seanbudd** [comment]
there's been a few issues filed with roole alert - can you check if this is a duplicate of any of thse: https://github.com/nvaccess/nvda/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+role+alert

**Comment by Adriani90** [comment]
@jcsteh any suggestion here? I think alerts are only read when an action is performed, but not when the page is loaded. Is aria errormessage more appropriate? for this use case?
